sql/src/main/java/org/apache/druid/sql/calcite/run/SqlResults.java

```diff
@@ -220,41 +221,51 @@ private static DateTime coerceDateTime(Object value, SqlTypeName sqlType)
     } else if (value instanceof DateTime) {
       dateTime = (DateTime) value;
     } else {
-      throw cannotCoerce(value, sqlType);
+      throw cannotCoerce(value, sqlType, fieldName);
     }
     return dateTime;
   }
 
   private static String coerceUsingObjectMapper(
       final ObjectMapper jsonMapper,
       final Object value,
-      final SqlTypeName sqlTypeName
+      final SqlTypeName sqlTypeName,
+      final String fieldName
   )
   {
     try {
       return jsonMapper.writeValueAsString(value);
     }
     catch (JsonProcessingException e) {
-      throw cannotCoerce(e, value, sqlTypeName);
+      throw cannotCoerce(e, value, sqlTypeName, fieldName);
     }
   }
 
-  private static IllegalStateException cannotCoerce(
+  private static DruidException cannotCoerce(
       final Throwable t,
       final Object value,
-      final SqlTypeName sqlTypeName
+      final SqlTypeName sqlTypeName,
+      final String fieldName
   )
   {
-    return new ISE(t, "Cannot coerce [%s] to [%s]", value == null ? "null" : value.getClass().getName(), sqlTypeName);
+    return DruidException.forPersona(DruidException.Persona.DEVELOPER)
```

Comment:
Hmm. I would say that this is for user persona since its an invalid input.

Reply:
fixed